msg/templates/uorb/microRTPS_client.cpp.template

```diff
@@ -0,0 +1,407 @@
+            // obtained data for the file descriptor
+            struct @(topic)_s data = {};
+            // copy raw data into local buffer
+            orb_copy(ORB_ID(@(topic)), fds[@(idx)], &data);
+            serialize_@(topic)(&data, data_buffer, &length, &microCDRWriter);
+            if (0 < (read = transport_node->write((char)@(message_id(topic)), data_buffer, length)))
+            {
+                total_sent += read;
+                ++sent;
+            }
+        }
+@[end for]@
+
+        usleep(_options.sleep_ms*1000);
+        ++loop;
+    }
+
+    struct timespec end;
+    px4_clock_gettime(CLOCK_REALTIME, &end);
+    double elapsed_secs = double(end.tv_sec - begin.tv_sec) + double(end.tv_nsec - begin.tv_nsec)/double(1000000000);
+    printf("\nSENT:     %" PRIu64 " messages in %d LOOPS, %" PRIu64 " bytes in %.03f seconds - %.02fKB/s\n",
+            sent, loop, total_sent, elapsed_secs, (double)total_sent/(1000*elapsed_secs));
+
+    return nullptr;
+}
+
+static int launch_send_thread(pthread_t &sender_thread)
+{
+    pthread_attr_t sender_thread_attr;
+    pthread_attr_init(&sender_thread_attr);
+    pthread_attr_setstacksize(&sender_thread_attr, PX4_STACK_ADJUSTED(4000));
+    struct sched_param param;
+    (void)pthread_attr_getschedparam(&sender_thread_attr, &param);
+    param.sched_priority = SCHED_PRIORITY_DEFAULT;
+    (void)pthread_attr_setschedparam(&sender_thread_attr, &param);
+    pthread_create(&sender_thread, &sender_thread_attr, send, nullptr);
+    pthread_attr_destroy(&sender_thread_attr);
+
+    return 0;
+}
+@[end if]@
+
+static int micrortps_start(int argc, char *argv[])
+{
+    if (0 > parse_options(argc, argv))
+    {
+        printf("EXITING...\n");
+        _rtps_task = -1;
+        return -1;
+    }
+
+    switch (_options.transport)
+    {
+        case options::eTransports::UART:
+        {
+            transport_node = new UART_node(_options.device, _options.baudrate, _options.poll_ms);
+            printf("\nUART transport: device: %s; baudrate: %d; sleep: %dms; poll: %dms\n\n",
+                   _options.device, _options.baudrate, _options.sleep_ms, _options.poll_ms);
+        }
+        break;
+        case options::eTransports::UDP:
+        {
+            transport_node = new UDP_node(_options.recv_port, _options.send_port);
+            printf("\nUDP transport: recv port: %u; send port: %u; sleep: %dms\n\n",
+                    _options.recv_port, _options.send_port, _options.sleep_ms);
+        }
+        break;
+        default:
+            _rtps_task = -1;
+            printf("EXITING...\n");
+        return -1;
+    }
+
+    if (0 > transport_node->init())
+    {
+        printf("EXITING...\n");
+        _rtps_task = -1;
+        return -1;
+    }
+
+    sleep(1);
+@[if recv_topics]@
+
+    char data_buffer[BUFFER_SIZE] = {};
+    int read = 0;
+    uint8_t topic_ID = 255;
+
+    // Declare received topics
+@[for topic in recv_topics]@
+    struct @(topic)_s @(topic)_data;
```

Comment:
Just to make it explicit: this makes the required stack size dependent on the number & set of enabled RTPS topics. From what I see the struct can just be moved inside the loop. This will then require less stack if multiple topics are configured.